### ISSUE TITLE
Invalid nested parameter format (3)

### ISSUE THREAD

**Issue by HolQue** [test-fullautomation/python-jsonpreprocessor#134]
The JSON code

```
{
   "ara"   : "ara",
   "param" : 123,
   "val1"  : ${p${ara}m}
}
```

causes the following error message:

`'Invalid nested parameter format: ${p${ara}m}, - The double quotes are missing!!!'`

Now I am thinking about. Because my first idea was was to get a parameter with name '`val1`' and value `123` (int!).

Because I want to keep the data type, I did not encapsulate the expression in quotes. This expression `${p${ara}m}` is not a composite string (that really would require quotes), it's a name of a single parameter, realized by nested other parameters.

When I read the error message, it seems to me, that the `${}` substitution inside expressions only works if the expression is encapsulated in quotes.

But when I do it in this way:

`"val1"  : "${p${ara}m}"`

I get a string value `'123'`. This is not what I want. I want an integer.

Could you please clarify: Is this format (nested parameters without quotes)

`"val1"  : ${p${ara}m}`

really invalid or is it possible to enable this format to have a chance to work also with other data types like strings?

**Comment by namsonx** [comment]
Hello Thomas,

Could you please give your comment regarding Holger's request?

Thank you,
Son

**Comment by test-fullautomation** [comment]
Hi Son,
puh... this is really difficult.
My feeling is that we should behave like a linux bash shell, whatever it is doing in this case.
String replacement proposed by Holger without quotes is also strange. What to do if key "ara" holds an integer?
Let us discuss this with Holger.
Thank you,
 Thomas

**Comment by namsonx** [comment]
Hello Holger,

As our discussion in the meeting, we temporary keep the behavior as it is, trying to find and follow a standard.

Thank you,
Son

**Comment by HolQue** [comment]
Again thinking about I would agree to keep the current behavior.

Retest:

```
"ara"   : "ara",
"param" : 123,
"val1"  : ${p${ara}m}
```

causes:

`Error: 'Invalid nested parameter format: ${p${ara}m} - The double quotes are missing!!!'!`

And we have to communicate the rule, that without quotes around the expression, a concatenation of strings based on parameters is not possible.

And with quotes:

```
"ara"   : "ara",
"param" : 123,
"val1"  : "${p${ara}m}"
```

it works fine.

`{'ara': 'ara', 'param': 123, 'val1': '123'}`

In my opinion the issue can be closed.

**Comment by namsonx** [comment]
merged to develop branch

**Comment by test-fullautomation** [comment]
solved with version 0.10.0
